### ISSUE TITLE
unnecessary-dunder-call doesn't understand mocks

### ISSUE THREAD

**Issue by kasium** [PyCQA/pylint#6823]
### Bug description

from unittest.mock import patch
import datetime

def under_test():
    datetime.datetime()()

def test():
    with patch.object(datetime, "datetime") as mock:
        under_test()
    mock.__call__().assert_called_once()

### Configuration

```ini
[tool.pylint."messages control"]
enable = ["all"]
```

### Command used

```shell
pylint foo.py
```

### Pylint output

```shell
************* Module foo
foo.py:10:4: C2801: Unnecessarily calls dunder method __call__. Invoke instance directly. (unnecessary-dunder-call)
```

### Expected behavior

Pylint should understand that we deal with a mock here

### Pylint version

```shell
pylint 2.14.0
astroid 2.11.5
Python 3.7.2 (default, Jun  2 2022, 07:23:31) 
[GCC 7.5.0]
```

### OS / Environment

SUSE Linux

**Comment by jacobtylerwalls** [comment]
Hey @kasium I just want to understand your report. Are you saying there is a reason the following code suggested by the pylint warning does not work or is not idiomatic?
```python
    mock().assert_called_once()
```
